lib/git.rb

```diff
@@ -127,6 +130,55 @@ def clone(remote, local, options = {})
       RubyGit.clone(remote, local, options)
     end
 
+    class RJGitSSHConfigCallback
+      include TransportConfigCallback
+      def initialize(private_key_file = nil, private_key_passphrase = nil)
+        @sshSessionFactory = Class.new(JschConfigSessionFactory) {
+          def initialize(private_key_file = nil, private_key_passphrase = nil)
+            super()
+            @private_key_file = private_key_file
+            @private_key_passphrase = private_key_passphrase
+          end
+
+          def configure(host, session)
+          end
+
+          def createDefaultJSch(fs)
+            default_j_sch = super(fs)
+            if @private_key_file
+              default_j_sch.removeAllIdentity()
+              if @private_key_passphrase
+                default_j_sch.addIdentity(@private_key_file, @private_key_passphrase)
+              else
+                default_j_sch.addIdentity(@private_key_file)
+              end
+            end
+            return default_j_sch
+          end
+        }.new(private_key_file, private_key_passphrase)
+      end
+
+      def configure(transport)
+        sshTransport = transport
+        sshTransport.setSshSessionFactory(@sshSessionFactory)
+      end
+    end
+
+    def self.set_command_transport(command, remote, options = {})
+      begin
+        uri = URI.parse(remote)
+      rescue URI::InvalidURIError
+        uri = nil
+      end
+      transport_protocol = uri.scheme if uri
+
+      if ("ssh" == transport_protocol) || (options[:transport_protocol] == :ssh) || options[:private_key_file]
```

Comment:
If the user has a .ssh/config file with a correct entry for the host of the remote in it, then it will succeed. This was my accidental first test case. :)